### PR TITLE
fix: [PL-38735]: NG reset password emails to go through NG

### PR DESCRIPTION
**Pull request by sahibaCCM**
## Describe your changes
 NG reset password emails to go through NG if the account is NG only or UI sends a flag isNG as true

Screen short of email template:
image

Tested for:
-> Sending emails through NG with isNG set as true.

## Checklist

https://github.com/harness/harness-core/assets/112424209/1ac659fb-2fe5-46e0-a548-713b80fda4f9

- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49758)

### CODE REVIEW

**Comment by sahibaCCM** [comment]
trigger messagecheck

**Comment by sahibaCCM** [comment]
trigger uts

**Comment by sahibaCCM** [comment]
trigger uts

**Comment by sahibaCCM** [comment]
trigger uts

**Comment by sahibaCCM** [comment]
trigger uts

**Comment by sahibaCCM** [comment]
trigger uts

**Comment by rijajoo** [comment]
UTs are failing.

**Comment by sahibaCCM** [comment]
trigger ut0

**Comment by sahibaCCM** [comment]
trigger uts

File: dockerization/manager/scripts/replace_configs.sh [hunk 1]
```diff
@@ -1221,3 +1221,6 @@ fi
 
 replace_key_value cdTsDbRetentionPeriodMonths "$CD_TSDB_RETENTION_PERIOD_MONTHS"
 replace_key_value enableOpentelemetry "$ENABLE_OPENTELEMETRY"
+replace_key_value notificationClient.httpClient.baseUrl "$NOTIFICATION_BASE_URL"
+replace_key_value notificationClient.secrets.notificationClientSecret "$NEXT_GEN_MANAGER_SECRET"
+replace_key_value notificationClient.messageBroker.uri "${NOTIFICATION_MONGO_URI//\\&/&}"
```
Comment (rijajoo) [inline dockerization/manager/scripts/replace_configs.sh:1226]: Is this a typo?

File: 220-graphql-test/src/supporter-test/java/io/harness/rule/GraphQLRule.java [hunk 2]
```diff
@@ -184,21 +207,43 @@ protected MainConfiguration getConfiguration(String dbName) {
                                             .connectTimeOutSeconds(15)
                                             .build())
             .build());
+    MarketPlaceConfig marketPlaceConfig =
```
Comment (rijajoo) [inline 220-graphql-test/src/supporter-test/java/io/harness/rule/GraphQLRule.java:210]: What's this block of code needed for?

**Review by rijajoo — CHANGES_REQUESTED** [review]

File: 220-graphql-test/src/supporter-test/java/io/harness/rule/GraphQLRule.java (hunk 2 above)
Comment (sahibaCCM) [reply 220-graphql-test/src/supporter-test/java/io/harness/rule/GraphQLRule.java:210]: the tests where failing due to missing configurations, due to the changes that where brought in by removing SpringPersistenceTestModule.

**Review by rijajoo — APPROVED** [review]

**Pull request merged**